### PR TITLE
Format merkle tree

### PR DESCRIPTION
**Pull request by hackaugusto**
## Describe your changes

Formatted result looks like this:

```
880abe452320966617646e7740b014954300f19a28780a0889d62ff33f4b0534
  1ade1369091efa31201e9b60c9c28874d0ddce5362b335135a6bb4c917285983
  3e60a9c843b4bb19f7a0572102e6507195f5240767a396335fd21981b048b807
    0100000000000000000000000000000000000000000000000000000000000000
    0200000000000000000000000000000000000000000000000000000000000000
    0300000000000000000000000000000000000000000000000000000000000000
    0400000000000000000000000000000000000000000000000000000000000000
```

and

```
[0400000000000000000000000000000000000000000000000000000000000000, 1ade1369091efa31201e9b60c9c28874d0ddce5362b335135a6bb4c917285983]
```

I added this to make it easier for me to visualize the data while debugging

## Checklist before requesting a review
- Repo forked and branch created from `next` according to naming convention.
- Commit messages and codestyle follow [conventions](./CONTRIBUTING.md).
- Relevant issues are linked in the PR description.
- Tests added for new functionality.
- Documentation/comments updated according to changes.

### CODE REVIEW

File: src/merkle/merkle_tree.rs [hunk 1]
```diff
@@ -154,6 +161,63 @@ impl MerkleTree {
...
+/// Converts a [Word] into hex.
+fn word_to_hex(w: &Word) -> Result<String, fmt::Error> {
+    let mut s = String::new();
+
+    for byte in w.iter().flat_map(|e| e.to_bytes()) {
+        write!(s, "{byte:02x}")?;
+    }
+
+    Ok(s)
+}
```
Comment (bobbinth) [inline src/merkle/merkle_tree.rs:173]: We already have a `to_hex()` utility in `miden-core` - should we use it here?

Comment (hackaugusto) [reply src/merkle/merkle_tree.rs:173]: I thought about it, this code is actually mostly based on that. The issue is that we would have a cyclic dependency, an alternative is to make miden-core use this instead, wdyt?

Comment (bobbinth) [reply src/merkle/merkle_tree.rs:173]: Ah - makes sense! I forgot that `miden-core` depends on `miden-crypto` and not the other way around.

I think it's fine to keep it in two places for now - but I wonder if we should move this one into a dedicated utils modules (e.g., maybe under `src/utils.rs`).

Comment (hackaugusto) [reply src/merkle/merkle_tree.rs]: moved

**Review by bobbinth — APPROVED** [review]
All looks good! Thank you!

**Pull request merged**